### PR TITLE
Feat(amp story page): Custom timezone support

### PR DESCRIPTION
**Pull request by Deokumarjnu**
# Description

Added support for custom timezone on amp pages for structure data.

Fixes # https://github.com/quintype/ace-planning/issues/463

### CODE REVIEW

File: server/amp/handlers/story-page.js [hunk 1]
```diff
@@ -67,8 +67,10 @@ async function ampStoryPageHandler(
     )
       return res.redirect(story.url);
 
+    const timezone = get(additionalConfig, ["publisher", "timezone"], null);
```
Comment (Athira001) [inline server/amp/handlers/story-page.js:70]: default timeZone is null?

Comment (Deokumarjnu) [reply server/amp/handlers/story-page.js:70]: Yes, the default date will look something like this `2021-07-12T03:51:41Z`

**Comment by ags1773** [comment]
@Deokumarjnu could you also pls add a unit test for this and documentation explaining that `timezone` has to be present in BK config for this to work

File: test/unit/amp/amp-story-page-handler.js [hunk 2]
```diff
@@ -263,9 +256,70 @@ describe("ampStoryPageHandler unit tests", function () {
...
+    );
+  });
+  it("Should render timezone if it's passed to seo if timezone is present in publisher config", async function () {
+    let seoPassedToAmpLib;
+    let timezone = null;
+    const dummySeo = (config, pageType) => {
+      return {
+        getMetaTags: (config, pageType, data = {}, url) => {
+          timezone = data.data.timezone;
+          return {
```
Comment (ags1773) [inline test/unit/amp/amp-story-page-handler.js:274]: can remove this return

File: test/unit/amp/amp-story-page-handler.js [hunk 3]
```diff
@@ -263,9 +256,70 @@ describe("ampStoryPageHandler unit tests", function () {
...
-    }))
+    assert.strictEqual(
+      additionalConfigReceivedByAmplib,
+      JSON.stringify({
+        key111: "this contains bk config by default",
+        key222: "this is the fetched additional config",
+      })
+    );
+  });
+  it("Should render timezone if it's passed to seo if timezone is present in publisher config", async function () {
```
Comment (ags1773) [inline test/unit/amp/amp-story-page-handler.js:267]: can change this to 
`should call seo lib with appropriate timezone if timezone info is present in publisher config, pass it on to amplib`

File: test/unit/amp/amp-story-page-handler.js [hunk 4]
```diff
@@ -263,9 +256,70 @@ describe("ampStoryPageHandler unit tests", function () {
...
+      domainSlug: null,
+      seo: dummySeo,
+      additionalConfig: { publisher: { timezone: "Asia/Kolkata" } },
+      ampLibrary: dummyAmpLib,
+      InfiniteScrollAmp: DummyInfiniteScrollAmp,
+    });
+    assert.strictEqual(timezone, "Asia/Kolkata");
+    assert.strictEqual(seoPassedToAmpLib, "2021-07-15T11:35:20.008+05:30");
+  });
+  it("Should render null if it's not passed to seo if timezone is not present in publisher config", async function () {
```
Comment (ags1773) [inline test/unit/amp/amp-story-page-handler.js:298]: and this to `should call seo lib with null timezone if timezone info is absent in publisher config`

**Review by ags1773 — APPROVED** [review]

**Pull request merged**